configtx/signer.go

```diff
@@ -48,7 +49,10 @@ func (s *SigningIdentity) Public() crypto.PublicKey {
 func (s *SigningIdentity) Sign(reader io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
 	switch pk := s.PrivateKey.(type) {
 	case *ecdsa.PrivateKey:
-		rr, ss, err := ecdsa.Sign(reader, pk, digest)
+		hasher := sha256.New()
```

Comment:
This will only work if we're using the P-256 curve. If we use P-384 we should use a hash that outputs 384 bits. Could we please introduce this? Look at how BCCSP does it in fabric proper.

Reply:
Following the lead of the msp signing identity code [here](https://github.com/hyperledger/fabric/blob/master/msp/identities.go#L216-L224), I've added SHA3_256 as well. 

While I see the BCCSP code implements other hashing functions, I can't tell if we truly support all of them.

Reply:
Fabric actually hardcodes the hash family to SHA2 and the hash function to SHA256: https://github.com/hyperledger/fabric/blob/master/msp/configbuilder.go#L337-L341

Reply:
Hmmm, unclear, because in core.yaml we can configure bccsp to use p-384 by setting security to 384...

https://github.com/hyperledger/fabric/blob/master/sampleconfig/core.yaml#L289

Reply:
Reverted this to simply use SHA-256 when hashing the message before signing, which is consistent with fabric.

Reply:
This is correct, while we have some shims for 384 bits, we ignore whatever you set it to in core.yaml and always generate 256

Reply:
>This is correct, while we have some shims for 384 bits, we ignore whatever you set it to in core.yaml and always generate 256

That's bad - great catch Will! I think this needs to be fixed - either by properly supporting p-384 or by removing this fake configurability.